agent/src/main/java/com/testsigma/agent/controllers/HomeController.java

```diff
@@ -84,4 +91,20 @@ public ResponseEntity<String> register(@PathVariable("uuid") String uuid,
     }
     return response;
   }
+  private Set<AgentBrowser> getUniqueBrowsersList(Set<AgentBrowser> browsers){
+    Set<AgentBrowser> set = browsers.stream().collect(Collectors.toSet());
+    Set<AgentBrowser> browsersList = new HashSet<>();
+    for(AgentBrowser set1 : set){
+      AgentBrowser temp1 = set1;
+       for(AgentBrowser set2 : set){
+         if(set1.getName()==set2.getName()){
+           if(temp1.getMajorVersion()<set2.getMajorVersion()){
```

Comment:
@nixonakash01 This can be done in a better way. Rather than having time complexity O(N^2), you can do it in O(N). Think about how you can extract the latest version among multiple duplicate ones in O(N)

Reply:
@tarun-testsigma please check latest commit